src/components/ColorPicker/components/ColorPickerContent/ColorPickerContentComponent.jsx

```diff
@@ -14,25 +15,67 @@ const ColorPickerContentComponent = ({
   noColorText,
   colorStyle,
   ColorIndicatorIcon,
+  SelectedIndicatorIcon,
   shouldRenderIndicatorWithoutBackground,
-  NoColorIcon
+  NoColorIcon,
+  isBlackListMode,
+  colorsList,
+  isMultiselect
 }) => {
   const onClearButton = useCallback(() => {
     onValueChange(null);
   }, [onValueChange]);
+
+  const [selectedColors, setSelectedColors] = useState(value);
+
+  const onColorClicked = useCallback(
+    color => {
+      if (!isMultiselect) {
+        onValueChange(color);
```

Comment:
you still have two separate concerns of updates.

please have only one useEffect to update - onValueChange.

The logic of what to send to the onValue change shoud also be there